### PR TITLE
expose propose_group_context_extensions and add a function

### LINKED ISSUE #1549: Expose GroupContextExtensionProposal API
For some reason, the API for proposing Group Context Extensions is currently `#[cfg(test)]`: https://github.com/openmls/openmls/blob/033f4e63d9b8d1bfe4377a341b8e9696f94712a2/openmls/src/group/mls_group/proposal.rs#L322-L353

This is wrong. Also, we should maybe add an integration test for group context extensions. That would have caught this accident.

### PR DESCRIPTION
**Pull request by franziskuskiefer**
Make propose_group_context_extensions not `test` only and add an integration test.

fixes #1549

### CODE REVIEW

File: openmls/tests/test_mls_group.rs [hunk 1]
```diff
@@ -1194,3 +1194,84 @@ fn mls_group_ratchet_tree_extension(ciphersuite: Ciphersuite, provider: &impl Op
         assert_eq!(error, WelcomeError::MissingRatchetTree);
     }
 }
+
+//
+// Test that the builder pattern accurately configures the new group.
```
Comment (kkohbrok) [inline openmls/tests/test_mls_group.rs:1199]: This description is not correct, is it?

**Review by kkohbrok — CHANGES_REQUESTED** [review]
One small nit, otherwise looks good!

File: openmls/tests/test_mls_group.rs (hunk 1 above)
Comment (franziskuskiefer) [reply openmls/tests/test_mls_group.rs]: yeah... c&p

**Review by kkohbrok — APPROVED** [review]
Thanks!

**Pull request merged**